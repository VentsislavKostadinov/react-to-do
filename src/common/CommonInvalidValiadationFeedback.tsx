import { Form } from 'react-bootstrap'
import { InvalidValiadationFeedback } from '../model/InvalidValiadationFeedbackProps.types'

export const CommonInvalidValiadationFeedback = ({
    text,
}: InvalidValiadationFeedback) => {
    return <Form.Control.Feedback type="invalid">{text}</Form.Control.Feedback>
}
