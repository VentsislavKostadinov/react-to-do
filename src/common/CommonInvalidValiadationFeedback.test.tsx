import React from 'react'
import { CommonInvalidValiadationFeedback } from './CommonInvalidValiadationFeedback'
import { CommonInvalidValiadationFeedbackProps } from '../model/CommonInvalidValiadationFeedbackProps.types'
import { render, screen } from '@testing-library/react'

describe('CommonInvalidValiadationFeedbackProps component', () => {
    const invalidFeedbackProps: CommonInvalidValiadationFeedbackProps = {
        text: 'Please add a description',
        dataTestid: 'invalid-feedback'
    }

    it('renders correctly', () => {
        const { asFragment } = render(
            <CommonInvalidValiadationFeedback
                {...invalidFeedbackProps}
            />,
        )

        expect(asFragment).toMatchSnapshot()
    })

    it('it corrects displayed', () => {
        render(
            <CommonInvalidValiadationFeedback
                {...invalidFeedbackProps}
            />,
        )

        const invalidFeedback = screen.getByTestId('invalid-feedback')
        expect(invalidFeedback).toHaveTextContent(invalidFeedbackProps.text)

    })
})
