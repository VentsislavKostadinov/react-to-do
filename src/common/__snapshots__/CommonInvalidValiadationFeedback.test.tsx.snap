// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonCommonInvalidValiadationFeedback component renders correctly 1`] = `[Function]`;

exports[`CommonInvalidValiadationFeedbackProps component renders correctly 1`] = `[Function]`;
